LOCK_WRITE enum value (in pal/inc/rt/objidl.h) conflicts with fcntl.h on Linux

Hi!

Is there anything we can do? @jkotas 

```
.../pal/inc/rt/objidl.h:106:7: error: expected identifier
    {   LOCK_WRITE      = 1,
        ^
/usr/include/x86_64-linux-gnu/bits/fcntl-linux.h:240:21: note: expanded from macro 'LOCK_WRITE'
# define LOCK_WRITE     128     /* ... which allows concurrent write operations.  */
                        ^
1 error generated.
```

--- a/src/pal/inc/rt/objidl.h
+++ b/src/pal/inc/rt/objidl.h
@@ -102,13 +102,6 @@ enum tagSTREAM_SEEK
     } 	STREAM_SEEK;
 
 typedef 
-enum tagLOCKTYPE
-    {	LOCK_WRITE	= 1,
-	LOCK_EXCLUSIVE	= 2,
-	LOCK_ONLYONCE	= 4
-    } 	LOCKTYPE;
-
-typedef 
 enum tagSTATFLAG
     {	STATFLAG_DEFAULT	= 0,
 	STATFLAG_NONAME	= 1,

